collector/xfrm.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !noxfrm
+// +build !noxfrm
+
+package collector
+
+import (
+	"fmt"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+)
+
+type xfrmCollector struct {
+	fs     procfs.FS
+	logger log.Logger
+}
+
+func init() {
+	registerCollector("xfrm", defaultEnabled, NewXfrmCollector)
```

Comment:
This is a bunch of extra cardinality, I'm not sure IPSec offload is popular enough to expose by default. 

I'm thinking this should be disabled by default since most users will just get 0 values for these metrics.

Reply:
I agree XFRM is only useful for a small percentage of users. I have changed it to be disabled by default and updated the readme.